browser-extension/mv3/src/client-scripts/ajaxRequestInterceptor.js

```diff
@@ -184,16 +184,37 @@ import { PUBLIC_NAMESPACE } from "common/constants";
     return responseModification.type === "static" && responseModification.serveWithoutRequest;
   };
 
-  const getFunctionFromCode = (code) => {
-    return new Function("args", `return (${code})(args);`);
-  };
+  const getFunctionFromCode = (() => {
+    let logShown = false;
+
+    return (code, ruleType) => {
+      try {
+        return new Function("args", `return (${code})(args);`);
+      } catch (e) {
+        notifyOnErrorOccurred({
+          initiatorDomain: location.origin,
+          url: location.href,
+        }).then(() => {
+          if (!logShown) {
+            logShown = true;
+            console.log(
+              `%cRequestly%c Please reload the page for ${ruleType} rule to take effect`,
+              "color: #3c89e8; padding: 1px 5px; border-radius: 4px; border: 1px solid #91caff;",
+              "color: red; font-style: italic"
+            );
+          }
+        });
+        return () => {};
+      }
+    };
+  })();
```

Comment:
Can we keep logShown outside the method. This looks weird read as the method is called at the same time